src/modules/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,301 @@
+      });
+    });
+
+    describe('and a dto with not found user provided', () => {
+      let signInDto: SignInDto;
+
+      beforeEach(async () => {
+        signInDto = {
+          email: 'notfounduser@gmail.com',
+          password: '123',
+        };
+
+        userRepository.findOne = jest.fn().mockReturnValueOnce(null);
+      });
+
+      it('should throw not found exception', async () => {
+        await expect(authService.signIn(signInDto)).rejects.toThrowError(
+          new NotFoundException(NOT_FOUND),
+        );
+      });
+    });
+
+    describe('and a dto with incorrect password provided', () => {
+      let signUpMock: SignUpDto;
+      let userDataMock: User;
+
+      beforeEach(async () => {
+        signUpMock = getSignUpData();
+        userDataMock = {
+          id: '1',
+          email: signUpMock.email,
+          phone: null,
+          firstName: signUpMock.firstName,
+          lastName: signUpMock.lastName,
+          password: signUpMock.password,
+          roles: ['customer'],
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        };
+
+        userRepository.findOne = jest.fn().mockReturnValueOnce(userDataMock);
+      });
+
+      it('should throw unauthorized exception', async () => {
+        const invalidSignIn: SignInDto = {
+          email: signUpMock.email,
+          password: 'invalid_credentials',
+        };
+
+        await expect(authService.signIn(invalidSignIn)).rejects.toThrowError(
+          new UnauthorizedException(INVALID_CREDENTIALS),
+        );
+      });
+    });
+  });
+
+  describe('when calling the refreshToken method', () => {
+    describe('and a valid refresh token provided', () => {
+      let tokensMock: Auth.AccessRefreshTokens;
+
+      beforeEach(async () => {
+        tokensMock = { accessToken: '123', refreshToken: '321' };
+        tokenService.refreshTokens = jest.fn().mockReturnValueOnce(tokensMock);
+      });
+
+      it('should return new tokens', async () => {
+        const token = 'valid_token';
+        expect(await authService.refreshTokens(token)).toBe(tokensMock);
+      });
+    });
+
+    describe('and a invalid refresh token provided', () => {
+      beforeEach(async () => {
+        tokenService.refreshTokens = jest
+          .fn()
+          .mockRejectedValueOnce(new UnauthorizedException());
+      });
+
+      it('should throw unauthorized exception', async () => {
+        const invalidRefreshToken = 'invalid_refresh_token';
+        await expect(
+          authService.refreshTokens(invalidRefreshToken),
+        ).rejects.toThrowError(new UnauthorizedException());
+      });
+    });
+  });
+
+  describe('when calling the logout method', () => {
+    describe('and valid arguments are provided', () => {
+      beforeEach(async () => {
+        tokenService.logout = jest.fn().mockReturnValueOnce(null);
+      });
+
+      it('should remove tokens from white list', async () => {
+        const userId = '1',
```

Comment:
Please check airbnb style guide for variable declaration https://github.com/airbnb/javascript#variables